src/runtime/virtcontainers/pkg/agent/protocols/client/client.go

```diff
@@ -345,7 +345,8 @@ func commonDialer(timeout time.Duration, dialFunc func() (net.Conn, error), time
 			return nil, timeoutErrMsg
 		}
 	case <-t.C:
-		cancel <- true
+		// Do not block, just notify the sub-goroutin to exit
+		close(cancel)
```

Comment:
I'd prefer fixing it by adding a buffer to the cancel channel. Something like
```
cancel := make(chan bool, 1)
```

Reply:
thanks, fixed!